Update  AE2 Fluid Crafting Addon Script

Mazlaci opened the pull request:
Expanded version of the AE2 Fluid Crafting addon script with balancing changes.

### Zenscript to Groovy
Converted the original zenscript file to groovy script.
I followed the structure of the original file, and referenced the original author since the old file is deleted.

### Recipe changes
Containt the following recipe changes and additions.

#### Extended Fluid Pattern Terminal
As per my earlier issue #1343 I updated the crafting revice to use Packaged Recipe Encoder instead of the Fluid Recipe Encoder, bringing it in line with the "normal" Extended Pattern Teminal
![image](https://github.com/user-attachments/assets/0be10fbc-a58c-4e48-8724-d19944c3b72c)

#### Ultimate Encoder
Originally ths file was unchanged using the default craft. This meant you could immediatly upgrade to an even bigger pattern encoder, before even meddling into PackagedAuto.
My design goal was to add it to the proper progression, giving you the option to use it later on, approximately around the time you unlock PackagedExCrafting.
The new recipe uses the Ultimate Catalyst. It means yue require Uranium to craft it. It's not exact but around the intended time, late midgame, early lategame when you can also make the Ultimate Extended Crafting Table.
Not only was this designwise what I intended, but it's thematic for the Ultimate Encoder to use the Ultimate Catalyst.
![image](https://github.com/user-attachments/assets/8b61e6ab-51b8-4f12-b67b-38336a795a49)

#### Fluid Level Maintainer
With both Lazy AE2 and AE2 Extended Fluid Crafting downloaded, you have access to the Level Maintainer and the Fluid Level Maintainer. They both serve the same purpose for fluids and items, but with the original recipe it meant Fluid Level Maintainer was avalable much earlier at HV while Item Level Maintainer required complex comonents around IV
I felt it gives for a better progression, if you unlock these later on not effetively immediatly after AE2.
I made the new Fluid Level Maintainer recipe based on the Item Maintainer, without using blocks from an another addon of course. It now needs az IV circuite and more complex alloys, meaning the similar finction is reflected in a similar crafting recipe.
![image](https://github.com/user-attachments/assets/7fdb5226-4837-4652-9227-9894770771e4)
For reference Item Level Maintainer recipe:
![image](https://github.com/user-attachments/assets/15865a36-e1b6-4958-87d4-b2bf803cd55f)

#### Fluid Assembler
This one is more of a thematic addition then anything. This recipe was also unchanged originally, but the default recipe needed iron ingots and green dye, which just felt very "un-gregtech-like". It now requires iron plates and 1k ME Fluid Components, meaning it's just a slight nerf for a more integrated recipe in the progression.
![image](https://github.com/user-attachments/assets/79689f12-1637-42e6-97de-f2c4d7c2f7da)

### Metaitem vs Item
In the case of applied energetics 2, for the life of me I cannot get metaitem to work...
Original file used item tags, so I used them here too, but added comment for every one of them for easy identification.

IntegerLimit commented on overrides/groovy/postInit/addon/AE2FluidCrafting.groovy:
```diff
@@ -0,0 +1,112 @@
+// MODS_LOADED: ae2fc
+
+// Original Compatibility Author: tracer4b in file AE2FluidCrafting.zs
```
We can remove this (since this is all one project and credits are only provided to those outside of the org).

Mazlaci replied:
Removed the unneeded credit

IntegerLimit commented on overrides/groovy/postInit/addon/AE2FluidCrafting.groovy:
```diff
@@ -0,0 +1,112 @@
+// MODS_LOADED: ae2fc
```
`// ^, Makes the script not run as long as AE2 FC isn't a mod in the player's instance.`

Please add this immediately beneath the `MODS_LOADED` comment, to clarify its purpose.

Mazlaci replied:
I added the requested explanation comment.

IntegerLimit commented on overrides/groovy/postInit/addon/AE2FluidCrafting.groovy:
```diff
@@ -0,0 +1,112 @@
+crafting.shapedBuilder().name('ae2fc:fluid_discretizer')
+	.output(item('ae2fc:fluid_discretizer'))
+	.matrix('EIE'
+		   ,'PGP'
```
The third level of indentation seems to be three spaces, it should still be a tab.

Mazlaci replied:
The reason I used spaces instead of a tab is to have the matrix rows under each other in an order.
I messed up while uploading the new file, using tab ident 8 instead of 4. Now if fix it, is it's as it should.

I can technically remove the space, but I would prefer to kepp it so it's visible at first glance.

IntegerLimit reviewed (changes requested):
Thank you for the PR! Always nice to see scripts converted to Groovy, and you have done this very well.

See comments for some small formatting changes required.

Also, since I don't have easy access to LazyAE2 instance, could you quickly state the crafting recipe for the item level maintainer? Thanks in advance.

IntegerLimit commented:
I would prefer if you used spaces as well for the last level of indentation; just to remain consistent with other files and our formatting specification (in `.editorconfig`).

Thanks in advance!

IntegerLimit reviewed (changes requested):
See above on relating to craft matrix formatting.

Mazlaci commented:
> See comments for some small formatting changes required.

I added the requested comment changes, and fixed the tab identation.
But as per my other comment the reason for 3 spaces is so the recipe matirx is aligned.

> Also, since I don't have easy access to LazyAE2 instance, could you quickly state the crafting recipe for the item level maintainer? Thanks in advance.

![image](https://github.com/user-attachments/assets/15865a36-e1b6-4958-87d4-b2bf803cd55f)

The recipe needs Fluix Steel Ingots, and Fluix Logic Unit
Tne Ingot is made from End Steel in the Alloy Smelter:r
![image](https://github.com/user-attachments/assets/a6ad12fd-26c7-4d56-9cbb-20326eb440dc)

The Logic Processor is a Circuit Assembler recipe with IV processor and titanium frame:
![image](https://github.com/user-attachments/assets/58413581-3a11-4ec8-8c4b-09208e225d0f)

IntegerLimit reviewed (approved):
LGTM. Thank you for correcting all the comments so quickly!
